### PR TITLE
[#WPLAT-953 Improvement] when select is present, select only those fields

### PR DESCRIPTION
**Pull request by tamassandor-planorama**
Ticket: https://planorama.atlassian.net/browse/WPLAT-953

### CODE REVIEW

File: src/api/lib/resources/lib/resource.js [hunk 1]
```diff
@@ -163,6 +163,18 @@ var Resource = NOOT.Object.extend(Authable).extend(Queryable).extend({
     callback = callback || stack.next;
     var pack = stack.package;
     if (!pack.messages.length) delete pack.messages;
+
+    // if there is a select qs, we override `minimize` settings
+    if (NOOT.tryGet(stack, 'query.raw.select')) {
+      var keys = _.chain(stack.query.select).keys().concat(['_id']).uniq().value();
+
+      pack.data = _.map(pack.data, function(item) {
+        var newItem = {};
```
Comment (peterporfy) [inline src/api/lib/resources/lib/resource.js]: `return _.pick(item, keys);`?

Comment (tamassandor-planorama) [inline src/api/lib/resources/lib/resource.js]: it didn't work for nested objects, like `pw.id` it actually wanted to find a key with this, so I had to add `tryGet` and `trySet`

Comment (peterporfy) [inline src/api/lib/resources/lib/resource.js]: ok

Comment (tamassandor-planorama) [inline src/api/lib/resources/lib/resource.js]: that would be out of scope now :D

**Pull request merged**